RFS/src/main/java/org/opensearch/migrations/bulkload/transformers/TransformerToIJsonTransformerAdapter.java

```diff
@@ -74,12 +76,21 @@ private static String printMap(Object map) {
     }
 
     @SuppressWarnings("unchecked")
-    private MigrationItem transformMigrationItem(MigrationItem migrationItem) {
+    private List<MigrationItem> transformMigrationItem(MigrationItem migrationItem) {
         // Keep untouched original for logging
         final Map<String, Object> originalMap = MAPPER.convertValue(migrationItem, Map.class);
-        Object transformedMigrationItem = transformer.transformJson(MAPPER.convertValue(migrationItem, Map.class));
-        logTransformation(originalMap, transformedMigrationItem);
-        return MAPPER.convertValue(transformedMigrationItem, MigrationItem.class);
+        Object transformedResult = transformer.transformJson(MAPPER.convertValue(migrationItem, Map.class));
+        logTransformation(originalMap, transformedResult);
+
+        List<MigrationItem> transformedItems = new ArrayList<>();
+        if (transformedResult instanceof List) {
+            for (Object item : (List<Object>) transformedResult) {
```

Comment:
nit: map/collect would work too - this and the next comment could return immutable lists.  If you can't do that, you should add why (as I think of it, you might not be able to do that - so it is tempting).